### PR TITLE
Added state and error_msg to bonsai-sdk

### PR DESCRIPTION
**Pull request by mothran**
This commit adds the `error_msg` and `state` members of the `SessionStatusRes` from the bonsai REST API.

These members provide insight into workflow failures and proving progress.

### CODE REVIEW

File: bonsai/sdk/src/alpha.rs [hunk 1]
```diff
@@ -95,6 +95,23 @@ pub mod responses {
...
         /// If the status == 'SUCCEEDED' then this should be present
         pub receipt_url: Option<String>,
+        /// Session Error message
+        ///
+        /// If the session is not RUNNING or SUCCEEDED, this is the error raised from within bonsai.
+        pub error_msg: Option<String>,
+        /// Session Proving State
+        ///
+        /// Indication of where in the proving pipeline the session currently is.
+        /// Possible states in order, include:
```
Comment (tzerrell) [inline bonsai/sdk/src/alpha.rs:105]: Perhaps indicate that this is `None` if the `status` isn't `RUNNING` (or does it also list state at exit in the case of `FAILED` / `TIMED_OUT` / `ABORTED`?)

Comment (tzerrell) [reply bonsai/sdk/src/alpha.rs:105]: Also this makes me notice that `SUCCEEDED` is double-listed on this line (and the line preceding it): https://github.com/risc0/risc0/pull/843/files#diff-9764b4c0c61823859f73ff49097c3ae82c7e8d9256e748ddaef9c66030e22df7R92

**Review by tzerrell — APPROVED** [review]
:tada::tada::tada: This is a much better experience, thank you!

**Review by flaub — APPROVED** [review]

File: bonsai/sdk/src/alpha.rs (hunk 1 above)
Comment (mothran) [reply bonsai/sdk/src/alpha.rs]: Thanks @tzerrell ! Good notes, I took a stab at refining it here: c63ebcf8a835a02733d6620a80987e9c434bc323 

Attached is a screenshot of what the formatting looks like. Thoughts?

**Review by tzerrell — APPROVED** [review]
Great, thank you!

**Pull request merged**